input/input.go

```diff
@@ -103,27 +116,40 @@ func (in DefaultHandler) ProcessMetricData(md *schema.MetricData, partition int3
 	err := md.Validate()
 	if err != nil {
 		in.invalidMD.Inc()
-		log.Debugf("in: Invalid metric %v: %s", md, err)
-
-		var reason string
-		switch err {
-		case schema.ErrInvalidIntervalzero:
-			reason = invalidInterval
-		case schema.ErrInvalidOrgIdzero:
-			reason = invalidOrgId
-		case schema.ErrInvalidEmptyName:
-			reason = invalidName
-		case schema.ErrInvalidMtype:
-			reason = invalidMtype
-		case schema.ErrInvalidTagFormat:
-			reason = invalidTagFormat
-		default:
-			reason = "unknown"
+
+		ignoreError := false
+		if err == schema.ErrInvalidTagFormat {
+			in.invalidTagMD.Inc()
+			if !rejectInvalidTags {
+				log.Debugf("in: Invalid metric %v, not rejecting it because rejection due to invalid tags is disabled: %s", md, err)
+				ignoreError = true
```

Comment:
this is fragile because md.Validate() just returns the first failure it encounters (out of possibly many). we rely on the implementation detail that in md.Validate(), the tag check is the last one.
it's possible that an md is invalid for multiple reasons. if we ever add another validation check after the tag check, we would ignore that other failure.

But i don't see a simple solution, so it's probably fine. maybe add a comment about this

Reply:
good point, will add a comment